Patch by MSP-Greg
Allows running Core RubyGems tests from installed Ruby
diff --git a/test/rubygems/test_gem.rb b/test/rubygems/test_gem.rb
index 2890e13dfb..ce0c4a79db 100644
--- a/test/rubygems/test_gem.rb
+++ b/test/rubygems/test_gem.rb
@@ -936,7 +936,7 @@ def test_self_platforms
   end
 
   def test_self_prefix
-    assert_equal PROJECT_DIR, Gem.prefix
+    assert_nil Gem.prefix
   end
 
   def test_self_prefix_libdir
@@ -1620,7 +1620,13 @@ def test_auto_activation_of_used_gemdeps_file
     assert_equal expected_specs, Gem.use_gemdeps.sort_by { |s| s.name }
   end
 
-  LIB_PATH = File.expand_path "../../../lib".dup.tap(&Gem::UNTAINT), __FILE__.dup.tap(&Gem::UNTAINT)
+  if defined? :RbConfig
+    t = RbConfig::CONFIG['rubylibdir']
+    LIB_PATH = File.exist?( File.join t, 'rubygems.rb') ? t :
+      File.expand_path("../../../lib".dup.tap(&Gem::UNTAINT), __FILE__.dup.tap(&Gem::UNTAINT))
+  else
+    LIB_PATH = File.expand_path("../../../lib".dup.tap(&Gem::UNTAINT), __FILE__.dup.tap(&Gem::UNTAINT))
+  end
   BUNDLER_LIB_PATH = File.expand_path $LOAD_PATH.find {|lp| File.file?(File.join(lp, "bundler.rb")) }.dup.tap(&Gem::UNTAINT)
   BUNDLER_FULL_NAME = "bundler-#{Bundler::VERSION}".freeze
 
diff --git a/test/rubygems/test_gem_commands_environment_command.rb b/test/rubygems/test_gem_commands_environment_command.rb
index 5dd0fc7521..e83c093cf9 100644
--- a/test/rubygems/test_gem_commands_environment_command.rb
+++ b/test/rubygems/test_gem_commands_environment_command.rb
@@ -26,7 +26,8 @@ def test_execute
     assert_match %r|RUBY VERSION: \d+\.\d+\.\d+ \(.*\) \[.*\]|, @ui.output
     assert_match %r|INSTALLATION DIRECTORY: #{Regexp.escape @gemhome}|,
                  @ui.output
-    assert_match %r|RUBYGEMS PREFIX: |, @ui.output
+# Gem.prefix is nil for ruby-loco
+#    assert_match %r|RUBYGEMS PREFIX: |, @ui.output
     assert_match %r|RUBY EXECUTABLE:.*#{RbConfig::CONFIG['ruby_install_name']}|,
                  @ui.output
     assert_match %r|GIT EXECUTABLE: #{@cmd.send(:git_path)}|, @ui.output
diff --git a/test/rubygems/test_gem_commands_update_command.rb b/test/rubygems/test_gem_commands_update_command.rb
index 340c658b9d..117d884d86 100644
--- a/test/rubygems/test_gem_commands_update_command.rb
+++ b/test/rubygems/test_gem_commands_update_command.rb
@@ -546,8 +546,11 @@ def test_update_rubygems_arguments
 
     arguments = @cmd.update_rubygems_arguments
 
-    assert_equal '--prefix',           arguments.shift
-    assert_equal Gem.prefix,           arguments.shift
+    # asserts fail from install folder
+    if Gem.method(:host).source_location[0].start_with? "#{File.expand_path '../../..', __FILE__}/"
+      assert_equal '--prefix',           arguments.shift
+      assert_equal Gem.prefix,           arguments.shift
+    end
     assert_equal '--no-document',      arguments.shift
     assert_equal '--previous-version', arguments.shift
     assert_equal Gem::VERSION,         arguments.shift
